RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -0,0 +1,198 @@
+    private var componentPaywallData: ComponentPaywallData
+
+    let configuration: TemplateViewConfiguration
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+    @Environment(\.locale)
+    var locale
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
+
+    init(_ configuration: TemplateViewConfiguration) {
+        self.configuration = configuration
+        self._componentPaywallData = .init(wrappedValue: .init(selectedPackage: configuration.packages.default))
+    }
+
+    var body: some View {
+        VStack(spacing: 0) {
+            if let data = self.configuration.components {
+                ComponentsView(
+                    locale: self.locale,
+                    components: data.components,
+                    configuration: self.configuration,
+                    shouldSplitLandscape: true
+                )
+                .environmentObject(self.componentPaywallData)
+            }
+        }
+        .edgesIgnoringSafeArea(.top)
+        .background(
+            try! PaywallColor(stringRepresentation: self.configuration.components!.backgroundColor.light).underlyingColor
+        )
+    }
+
+}
+
+func getLocalization(_ locale: Locale, _ displayString: DisplayString) -> String {
+    if let found = displayString.value[locale.identifier] {
+        return found
+    }
+
+    return displayString.value.values.first!
+}
+
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct ComponentsView: View {
+
+    let locale: Locale
+    let components: [PaywallComponent]
+    let configuration: TemplateViewConfiguration
+    let shouldSplitLandscape: Bool
+
+    init(locale: Locale, components: [PaywallComponent], configuration: TemplateViewConfiguration, shouldSplitLandscape: Bool = false) {
+        self.locale = locale
+        self.components = components
+        self.configuration = configuration
+        self.shouldSplitLandscape = shouldSplitLandscape
+    }
+
+    var landscapeLeftComponents: [PaywallComponent] {
+        return self.components.filter { component in
+            guard let displayPreferences = component.displayPreferences else {
+                return true
+            }
+            return displayPreferences.contains(.landscapeLeft)
+        }
+    }
+
+    var landscapeRightComponents: [PaywallComponent] {
+        return self.components.filter { component in
+            guard let displayPreferences = component.displayPreferences else {
+                return true
+            }
+            return displayPreferences.contains(.landscapeRight)
+        }
+    }
+
+    var portraitComponents: [PaywallComponent] {
+        return self.components.filter { component in
+            guard let displayPreferences = component.displayPreferences else {
+                return true
+            }
+            return displayPreferences.contains(.portrait)
+        }
+    }
```

Comment:
Note for later: this was done _mainly_ for proof of conept. We will be having more discussions about how to properly handle landscape